src/Core/Generator/Sampler/CosmosExecutor.cs

```diff
@@ -0,0 +1,114 @@
+    /// </summary>
+    internal class CosmosExecutor
+    {
+        private Container _container;
+        private ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CosmosExecutor"/> class.
+        /// </summary>
+        /// <param name="container">The Azure Cosmos DB container instance to interact with.</param>
+        /// <param name="logger">The logger instance used for logging information and debugging messages.</param>
+        public CosmosExecutor(Container container, ILogger logger)
+        {
+            this._container = container;
+            this._logger = logger;
+        }
+
+        /// <summary>
+        /// Executes the specified query on the Azure Cosmos DB container and returns the results as a list of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The type to which the query results should be deserialized.</typeparam>
+        /// <param name="query">The SQL-like query string to execute against the Azure Cosmos DB container.</param>
+        /// <param name="callback">Optional. A callback function that can be used to manipulate or process each retrieved item.</param>
+        /// <returns>A task representing the asynchronous operation, containing a list of results of type <typeparamref name="T"/>.</returns>
+        /// <exception cref="Exception">Thrown when the query execution fails with an error message and status code.</exception>
+        public async Task<List<T>> ExecuteQueryAsync<T>(string query, Action<T?>? callback = null)
+        {
+            _logger.LogDebug("Executing Query: {0}", query);
+
+            List<T> dataArray = new();
+
+            FeedIterator queryIterator = _container.GetItemQueryStreamIterator(new QueryDefinition(query));
+            while (queryIterator.HasMoreResults)
+            {
+                ResponseMessage item = await queryIterator.ReadNextAsync();
+
+                if (item.IsSuccessStatusCode)
+                {
+                    using StreamReader sr = new(item.Content);
+                    string content = await sr.ReadToEndAsync();
+
+                    JsonDocument jsonDocument = JsonDocument.Parse(content);
+                    JsonElement root = jsonDocument.RootElement.GetProperty("Documents");
+
+                    if (root.ValueKind == JsonValueKind.Array)
+                    {
+                        foreach (JsonElement element in root.EnumerateArray())
+                        {
+                            Process(callback, dataArray, element);
+                        }
+                    }
+                    else
+                    {
+                        Process(callback, dataArray, root);
+                    }
+                }
+                else
+                {
+                    throw new Exception(string.Format("Failed to execute query: {0} with status code {1}, Error Message : {2}", query, item.StatusCode, item.ErrorMessage));
```

Comment:
where is this caught? The code which references this function doesn't use try/catch. Is it helpful to print the whole exception to the console for the user? I suppose just the error message you're creating here is sufficient.